.github/workflows/create-release.yml

```diff
@@ -0,0 +1,129 @@
+      # Its value will be set by the action (if release-like tag is created)
+      VERSION: ""
+      # Its value will be set by the action (if release-like tag is created)
+      PRERELEASE: ""
+      # The version of PHP to be used here
+      INSTALLER_PHP_VERSION: '7.3'
+      # The minimum PHP version the built phar should support
+      PHAR_PHP_VERSION: '7.1'
+      # The version of Box to be used
+      BOX_VERSION: 3.13.0
+    steps:
+      - name: Check tag format
+        if: github.event_name == 'create' && github.event.ref_type == 'tag'
+        run: |
+          VERSION="$(printf '%s' "$GITHUB_REF" | sed -E 's/^refs\/tags\/v?([0-9]+\.[0-9]+\.[0-9]+.*)$/\1/')"
+          if printf '%s' "$VERSION" | grep -Eq '^[0-9]+\.[0-9]+\.[0-9]'; then
+            case "$VERSION" in
+              *dev* | *snap* | *[Aa]* | *[Bb]* | *[Rr][Cc]*)
+                PRERELEASE=true
+                ;;
+              *)
+                PRERELEASE=false
+                ;;
+            esac
+            printf 'The tag %s is for version %s\n' "${GITHUB_REF#refs/tags/}" "$VERSION"
+            printf 'Pre-release: %s\n' "$PRERELEASE"
+            printf 'VERSION=%s\n' "$VERSION" >> "$GITHUB_ENV"
+            printf 'PRERELEASE=%s\n' "$PRERELEASE" >> "$GITHUB_ENV"
+          else
+            printf 'The ref %s is not for a version\n' "$GITHUB_REF"
+          fi
+      - name: Checkout
+        if: env.VERSION != ''
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup PHP
+        if: env.VERSION != ''
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ env.INSTALLER_PHP_VERSION }}
+          tools: composer:v2
+          coverage: none
+          ini-values: phar.readonly=0
+      - name: Configure environment
+        if: env.VERSION != ''
+        run: composer config platform.php "$PHAR_PHP_VERSION"
+      - name: Install Composer dependencies
+        if: env.VERSION != ''
+        run: composer update --no-dev --no-progress --optimize-autoloader --ansi --no-interaction --no-cache
+      - name: Download Box
+        if: env.VERSION != ''
+        run: curl -Lf -o box.phar "https://github.com/box-project/box/releases/download/$BOX_VERSION/box.phar"
+      - name: Create the PHAR file
+        if: env.VERSION != ''
+        run: php box.phar compile --ansi --no-interaction
+      - name: Check if the PHAR file works
+        if: env.VERSION != ''
+        run: ./concrete.phar list --ansi
+      - name: Create the signature file
+        if: env.VERSION != ''
+        run: |
+          HASH="$(sha384sum concrete.phar | cut -f1 -d' ')"
+          printf 'Calculated signature: %s\n' "$HASH"
+          printf '%s' "$HASH" > concrete.sig
+      - name: Create the version file
+        if: env.VERSION != ''
+        run: printf '%s' "$VERSION" > version.txt
+      - name: Build release notes
+        if: env.VERSION != ''
+        run: |
+          CURRENT_TAG_FOUND=n
```

Comment:
Would it be worth putting this in a bash script underneath .github? IIRC it can still be called, the only value I'm interested in is syntax highlighting

Reply:
Done: 2880cd3be190fd13a16e779dcf7132e4e3334dae